### ISSUE TITLE
[data grid] How to position the columns panel from the gridtoolbar towards the right side when the custom button is towards the right

### ISSUE THREAD

**Issue by spoorti22** [mui/mui-x#10396]
Reference issue: I have a custom button towards the right side that loads only the column panel from the grid tool bar however I want it to position right below the custom button. 
https://github.com/mui/mui-x/issues/10342

**Comment by spoorti22** [comment]
@cherniavskii can you please help with this since you initially helped with the issue and have context of the problem?

**Comment by spoorti22** [comment]
Also since there is no explicit close button for the column panel when we click away since the popper panel has no onclick away listener defined how do we handle graceful close?

**Comment by cherniavskii** [comment]
Hi @spoorti22 
Hmm, this is a bit tricky, I'll try to come up with a solution later

**Comment by cherniavskii** [comment]
Hi @spoorti22
I tried utilizing [`anchorEl` callback signature](https://mui.com/material-ui/api/popper/#Popper-prop-anchorEl) and it seems to work: https://codesandbox.io/s/runtime-architecture-jytjx7?file=/Demo.tsx

Is this the desired behavior in your use case?

**Comment by spoorti22** [comment]
@cherniavskii Yeah this works, how do we handle the custom css for the panel popup to place it a bit below the button?

**Comment by tangentlin** [comment]
> Hi [@spoorti22](https://github.com/spoorti22) I tried utilizing [`anchorEl` callback signature](https://mui.com/material-ui/api/popper/#Popper-prop-anchorEl) and it seems to work: https://codesandbox.io/s/runtime-architecture-jytjx7?file=/Demo.tsx
> 
> Is this the desired behavior in your use case?

This works a like a charm, the only issue I have run into is TypeScript complains about the absence of `anchorEl` in type definition.

**Comment by michelengelen** [comment]
Following up since the API shifted since this thread. The `anchorEl` + `open` combination from the earlier answer predates a rename: the `panel` slot now exposes `target` instead of `anchorEl` for overriding the anchor element, which is exactly why TypeScript started complaining about `anchorEl` for anyone still using it.

```tsx
<DataGrid slotProps={{ panel: { target: buttonRef.current } }} />
```

This is the same anchor mechanism used by the column menu internally, so it should support the callback-based dynamic positioning from cherniavskii's example too, just with `target` instead of `anchorEl`.

Closing as answered. If `target` doesn't behave the same way as the old `anchorEl` callback on the latest version, please open a new issue with a reproduction.
